### DISCUSSION TITLE
How to solve filament export doesn’t work in laravel 11

### DISCUSSION THREAD

**Discussion by christmex** [filamentphp/filament#13264] [General]
In case someone needs this,

by default, laravel 11 will set your QUEUE_CONNECTION to database, that's why it doesn't work until you run the ``` php artisan queue:work ``` in my case, so I need to change the QUEUE_CONNECTION  to sync then restart my server. after that, it works.

Thank you.

**Comment by H-Hp** [comment]
Hi
Thank you for sharing this valuable information. It's indeed a common issue that developers might encounter when working with queues in Laravel, especially when upgrading to a newer version.

For the benefit of others who might face a similar situation, let's summarize the key points:

1. **Default Queue Connection in Laravel 11**
   By default, in Laravel 11, the `QUEUE_CONNECTION` environment variable is set to `database`. This means that Laravel expects you to have a database table set up to handle queued jobs.

2. **Running the Queue Worker**
   If you don't run the `php artisan queue:work` command, which starts the queue worker process, the queued jobs won't be processed. This is because Laravel assumes you want to use a separate process (the queue worker) to handle the queued jobs asynchronously.

3. **Changing the Queue Connection to `sync`**
   As you mentioned, changing the `QUEUE_CONNECTION` environment variable to `sync` can resolve the issue. With this setting, Laravel will process queued jobs synchronously, meaning the jobs will be executed immediately instead of being pushed to a queue.

4. **Restarting the Server**
   After changing the `QUEUE_CONNECTION` value, you need to restart your server or application for the new configuration to take effect.

By setting `QUEUE_CONNECTION=sync`, you don't need to run the `php artisan queue:work` command separately, as the queued jobs will be processed immediately during the request lifecycle.

It's important to note that using the `sync` queue driver is generally recommended for development and testing purposes only. In production environments, it's advisable to use a more robust queue driver like `database`, `redis`, or `sqs`, and run the `php artisan queue:work` command to ensure that queued jobs are processed reliably and asynchronously.

Thank you again for sharing this solution. It will undoubtedly help others who face similar issues when working with queues in Laravel 11 or other versions.

**Reply by muhreezky** [reply]
I changed the QUEUE_CONNECTION to 'sync', but when I click the export button, it shows error 419 | Page expired. How to deal with this? I use filament v3.3 with laravel 11 on php 8.2

**Reply by micraux** [reply]
Restart your server?

**Comment by kwaliyo** [comment]
Did QUEUE_CONNECTION to 'sync', the issue is the notification icon doesn't show the message badge, even if i click to open the notification pane no notification is there. The exports works but for the above mentioned issue.

**Reply by christmex** [reply]
did you activate the database notification?

https://filamentphp.com/docs/3.x/panels/notifications

**Reply by kwaliyo** [reply]
> did you activate the database notification?
> 
> https://filamentphp.com/docs/3.x/panels/notifications

yes I did, I do get a pop-up notification that my export is ready to download. But the notification pane is empty

**Reply by christmex** [reply]
did u check the database? is it exist?

**Reply by kwaliyo** [reply]
sure, when I changed  QUEUE_CONNECTION to 'database', and start the queue manually, works, but i dont want to manually run the queue manually everytime

**Reply by christmex** [reply]
i mean it should be in your database even tho u use sync, sync is directly run not using queue, I just want to make sure the notification is exist in database, if the notification exist in database then something wrong with the app, but if the data is not in ur database then something wrong with the process

**Reply by kwaliyo** [reply]
> i mean it should be in your database even tho u use sync, sync is directly run not using queue, I just want to make sure the notification is exist in database, if the notification exist in database then something wrong with the app, but if the data is not in ur database then something wrong with the process

All necessary tables exits. Done export import before, this is my first time encountering this kind of issue

**Reply by christmex** [reply]
i understand all the table exist, but I mean the data in that table, the notification is stored on that table, so u need to make sure if the notification data is already there, so if the notification data is not in that table then something is wrong, but if the notification data exist so something sis wrong with the app, that's how I debug something 😁

**Reply by christmex** [reply]
so like when ure done with export stuff then check the data in notifications table, is it the notification exist or not

**Reply by kwaliyo** [reply]
yes.

**Reply by christmex** [reply]
maybe with clear cache? 

php artisan cache:clear && php artisan optimize:clear

also, please show us ur adminpanel service provider

**Reply by kwaliyo** [reply]
> maybe with clear cache?
> 
> php artisan cache:clear && php artisan optimize:clear
> 
> also, please show us ur adminpanel service provider
![Screenshot 2025-05-24 092707](https://github.com/user-attachments/assets/bb5e27a9-99ba-4026-a448-4999b77c9cf9)
![Screenshot 2025-05-24 092453](https://github.com/user-attachments/assets/884c6b46-6af1-40e3-bba3-c17962b93a19)
![Screenshot 2025-05-24 092433](https://github.com/user-attachments/assets/c17356ef-5f8e-456e-84ab-5e63efba09e3)

**Reply by kwaliyo** [reply]
![image](https://github.com/user-attachments/assets/49c92340-6610-4ce4-a178-22027de33bfc)
![Screenshot 2025-05-24 104148](https://github.com/user-attachments/assets/9b98b593-e411-406e-b37d-c17edf455c86)

**Reply by christmex** [reply]
when the export completed, please open the notifications table to see if there is data inside that table, cuz I saw it was 0 byte, which is mean there is nothing inside, I just want to make sure, the notification saved

**Reply by kwaliyo** [reply]
![image](https://github.com/user-attachments/assets/9d2d5cff-8e21-4138-8c31-ca62433c8a64)
Indeed, it is empty

**Reply by christmex** [reply]
thats why in your notification panel is empty, cuz in the database is also empty

**Reply by kwaliyo** [reply]
> thats why in your notification panel is empty, cuz in the database is also empty

its obvious, what might be likely causes?

**Reply by christmex** [reply]
im sorry im trying to figure it out, but still have no clue, sometimes I need to get my hands on it to find the problem, but maybe ur project is not for public

**Reply by christmex** [reply]
i was facing the same problem but that's works when I change the QUEU_DATABASE value, it mybe u can try set up again to database and run the PHP artisan queu:work just to see if the notification works

**Reply by ialtoobi** [reply]
> ![image](https://private-user-images.githubusercontent.com/65981875/447216023-49c92340-6610-4ce4-a178-22027de33bfc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU0MzQ3MzMsIm5iZiI6MTc1NTQzNDQzMywicGF0aCI6Ii82NTk4MTg3NS80NDcyMTYwMjMtNDljOTIzNDAtNjYxMC00Y2U0LWExNzgtMjIwMjdkZTMzYmZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE3VDEyNDAzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0ZTk4YTMzZWE0MjMzMzRkZDk4OWYxNDU5MWY2NWY4NGRmNmY3MmU5OGRkY2ZjOWFiYjAxN2ZlMzUxODY5ZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SPh-kJgpKqk-QStlQEcfepP161Tvj_GMboNtZ7pCMgY) ![Screenshot 2025-05-24 104148](https://private-user-images.githubusercontent.com/65981875/447216042-9b98b593-e411-406e-b37d-c17edf455c86.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU0MzQ3MzMsIm5iZiI6MTc1NTQzNDQzMywicGF0aCI6Ii82NTk4MTg3NS80NDcyMTYwNDItOWI5OGI1OTMtZTQxMS00MDZlLWIzN2QtYzE3ZWRmNDU1Yzg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE3VDEyNDAzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0YTk2MWYwZjUxY2JhY2ZjMGI3NDBjNmNmNWRmN2MzYjY3MDlmZjE5NzY0YjYxZWVlZWNlYWM0YWUwMzViMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-KHzsWDUJX-d7e_G5kjCRs4okI1Xsk1QgJIOF7zzeLo)

did you fix this issue?.. I'm facing same issue here!

**Reply by christmex** [reply]
@ialtoobi Yeah, if you are in development, just change the value to sync, then it should be working

**Reply by ialtoobi** [reply]
It's working with sync, but database notifications doesn't work
